File: lib/data/remote/model/music_group_dto.dart
import 'package:json_annotation/json_annotation.dart';

part '../../../generated/data/remote/model/music_group_dto.g.dart';

@JsonSerializable(createToJson: false)
class MusicGroupDTO {
  final int id;
  final String? title;
  final String? history;
  final String? leader;
  final String? image;
  final bool isActive;

  MusicGroupDTO({
    required this.id,
    required this.title,
    required this.history,
    required this.leader,
    required this.image,
    required this.isActive,
  });

  static const fromJsonFactory = _$MusicGroupDTOFromJson;

  @override
  String toString() {
    return 'MusicGroupDTO{id: $id, title: $title, history: $history, leader: $leader, image: $image, isActive: $isActive}';
  }
}

File: composeApp/src/commonMain/kotlin/by/geth/gethsemane/data/source/remote/model/MusicGroupDTO.kt
package by.geth.gethsemane.data.source.remote.model

import kotlinx.serialization.SerialName
import kotlinx.serialization.Serializable

@Serializable
data class MusicGroupDTO(
    @SerialName("id")
    val id: Int,
    @SerialName("title")
    val title: String,
    @SerialName("history")
    val history: String? = null,
    @SerialName("leader")
    val leader: String? = null,
    @SerialName("image")
    val image: String? = null,
    @SerialName("isActive")
    val isActive: Boolean,
)
